test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using Microsoft.SqlTools.ServiceLayer.SchemaCompare;
+using Xunit;
+
+namespace Microsoft.SqlTools.ServiceLayer.UnitTests.SchemaCompare
+{
+    public class SchemaCompareTests
+    {
+        [Fact]
+        public void FormatScriptAddsGo()
+        {
+            string script = "EXECUTE sp_addextendedproperty @name = N'MS_Description', @value = N'Primary key for AWBuildVersion records.', @level0type = N'SCHEMA', @level0name = N'dbo', @level1type = N'TABLE', @level1name = N'AWBuildVersion', @level2type = N'COLUMN', @level2name = N'SystemInformationID';";
+            Assert.DoesNotContain("GO", script);
+            string result = SchemaCompareOperation.FormatScript(script);
+            Assert.Contains("GO", result);
+            Assert.EndsWith("GO", result);
+        }
+
+        [Fact]
+        public void FormatScriptDoesNotAddGoForNullScripts()
+        {
+            string script1 = null;
+            string result1 = SchemaCompareOperation.FormatScript(script1);
+            Assert.DoesNotContain("GO", result1);
+            Assert.Equal(null, result1);
+
+            string script2 = "null";
+            string result2 = SchemaCompareOperation.FormatScript(script2);
+            Assert.DoesNotContain("GO", result2);
+        }
+
+        [Fact]
+        public void FormatScriptDoesNotAddGoForEmptyStringScripts()
+        {
+            string script = string.Empty;
+            string result = SchemaCompareOperation.FormatScript(script);
+            Assert.DoesNotContain("GO", result);
+            Assert.Equal(string.Empty, result);
+        }
+
+        [Fact]
+        public void FormatScriptDoesNotAddGoForWhitespaceStringScripts()
+        {
+            string script = "    \t\n";
+            Assert.True(string.IsNullOrWhiteSpace(script));
+            string result = SchemaCompareOperation.FormatScript(script);
+            Assert.DoesNotContain("GO", result);
+            Assert.Equal(string.Empty, result);
+        }
+
+        [Fact]
+        public void RemovesExcessWhitespace()
+        {
+            // leading whitespace
+            string script1 = "\r\nEXECUTE sp_addextendedproperty @name = N'MS_Description', @value = N'Primary key for AWBuildVersion records.', @level0type = N'SCHEMA', @level0name = N'dbo', @level1type = N'TABLE', @level1name = N'AWBuildVersion', @level2type = N'COLUMN', @level2name = N'SystemInformationID';";
+            string result1 = SchemaCompareOperation.RemoveExcessWhitespace(script1);
```

Comment:
I'd also add a test that has actual whitespace, not just newlines at the beginning/end

Reply:
added a couple spaces to these tests to make sure those get removed too